### ISSUE TITLE
Why not exiftool?

### ISSUE THREAD

**Issue by UnixCro** [wez/atomicparsley#57]
Hello,
only from curiosity. No bugs or anything like that I have to report. But isn't what AtomicParsley can't already do with exiftool? Why should one rather use AtomicParsley?

Best regards
UnixCro

**Comment by wez** [comment]
Shrug; it's up to you to compare and decide which you want to use.

**Comment by UnixCro** [comment]
Yes but I am not familiar with such things and you are a developer who could develop something complicated.

Only now I've been given the task of editing metadata and the best ones I found were yours, eyeD3 and exiftool but which one should I use? Do you have any advantages over exiftool.

I apologize for asking this question, but please answer.

Best regards
UnixCro

**Comment by wez** [comment]
I have not used the others, and I don't actively use this project: https://github.com/wez/atomicparsley#a-note-on-maintenance
You're asking me to consult and weigh in on a technical matter and I simply do not have the bandwidth or motivation to do so.

**Comment by UnixCro** [comment]
It's really weird to build something without ever checking how others did it. But that's the situation...

Even if all this seems very strange to me, I am still fascinated by the program even if I somehow do not think that you are the author of this program from your contributions.

I've now completely memorized `AtomicParsley` as it helps me with my work and admittedly the only program I could currently find that is capable of setting a cover in `M4A` or setting track/disc et cetera ... in these files.


`exiftool` is an excellent program for extracting metadata, but it is very limited when it comes to `MP4/M4a` editing. So for editing I will be using `AtomicParsley`. `eyeD3` is currently the best in terms of `MP3` editing.

Thank you for your time and for letting me get to know you.

**Comment by wez** [comment]
> It's really weird to build something without ever checking how others did it. But that's the situation...

You should check your condescending tone; you're making a lot of assumptions about who I am, what I did and how I got here.

Your tone is judgmental and the foundation of it is misplaced. You should take the time to educate yourself on the history of a project before leaping to conclusions about the people currently involved, and even then, it is often better to keep those judgements to yourself because you still won't have the whole story.

None of this is relevant to the question you asked, so all you have achieved with making such a statement is being offensive.

**Comment by malhal** [comment]
exiftool has bugs like it displays the artist atom atID as album title.
It also incorrectly formats the album ID plID as 8 bytes instead of an integer.
```
Album Title ID                  : 450757148 // this is the artist ID atID
Play List ID                    : 0 0 0 0 90 117 21 79 // this should be 1517622607 the album ID
```

I'd be interested to know how to write these `atID` and `plID` atoms using AtomicParsely I don't see it mentioned in the help anywhere.

**Comment by seabopo** [comment]
> exiftool has bugs like it displays the artist atom atID as album title. It also incorrectly formats the album ID plID as 8 bytes instead of an integer.
> 
> ```
> Album Title ID                  : 450757148 // this is the artist ID atID
> Play List ID                    : 0 0 0 0 90 117 21 79 // this should be 1517622607 the album ID
> ```
> 
> I'd be interested to know how to write these `atID` and `plID` atoms using AtomicParsely I don't see it mentioned in the help anywhere.

Hi Malcolm,

Did you ever figure out how to write the `atID` and `plID` atoms using AtomicParsely? 

I've spent hours looking for a way for those and a few other tags and cannot find a way to do so.

**Comment by malhal** [comment]
Yep in my script first I get all the fields:
```
ATOMIC=$(/opt/homebrew/bin/AtomicParsley "file" -t)
```
Then I get the `atID` like:
```
ARTISTID=$(echo "$ATOMIC" | grep atID | cut -d " " -f4-)
```
And `plID` like:
```
ALBUMID=$(echo "$ATOMIC" | grep plID | cut -d " " -f4-)
```
AtomicParsley fails if the path to the file is too long so often I have to rename the file to say `a.m4a` before running it then rename it back which is super annoying.

**Comment by seabopo** [comment]
Hi Malcolm,

Thank you for responding so quickly. 

Sorry if I wasn't clear. I have no problem reading those atoms ... I was trying to figure out how to write them back to the file. 

There is no AtomicParsley parameter documented in the help. I tried writing them back using the atomIDs since apID, geID and cnID are supported, just in case they worked but weren't documented, but got errors. 

I tried writing them back as iTunes rDNS atoms and UUID atoms in the hopes AtomicParsley was doing something special for them, but that doesn't seem to be the case. When I re-read the updated file the atoms aren't the same as the originals and conform to the rDNS and UUID forms.

I was hoping you had found how to write those atoms. I've been looking through the code for a 'manualAtomAdd' equivalent, but haven't found one yet.
